### PR TITLE
[Poisson Reconstruction] Compatibility with Simple Cartesian

### PR DESCRIPTION
**Pull request by sgiraudot**
## Summary of Changes

Poisson currently only compiles with Epick (see https://github.com/CGAL/cgal/issues/5431). This PR makes it compile with simple cartesian (both using double and float). However, the code rarely works at runtime (because it is based on Delaunay triangulation which are not robust without exact predicates), so I couldn't add a test for this.

I have tried to make it compile with Epeck, but it required many changes (including changes in Surface Mesher) and ended up breaking the Epick version, so I figured it was not worth the trouble.

## Release Management

* Affected package(s): Poisson
* Issue(s) solved (if any): fix https://github.com/CGAL/cgal/issues/5431

### CODE REVIEW

File: Number_types/include/CGAL/NT_converter.h [hunk 1]
```diff
@@ -72,6 +83,17 @@ struct NT_converter < double, double >
     }
 };
 
+template <>
+struct NT_converter < float, float >
```
Comment (lrineau) [inline Number_types/include/CGAL/NT_converter.h:87]: Is not there already `template <class NT1> struct NT_converter<NT1, NT1>`?!

Comment (sgiraudot) [reply Number_types/include/CGAL/NT_converter.h:87]: I just had a look: you are correct, but for some reason there is also a specialization for `double/double` (which is why I introduced the `float/float` one). Both could be removed, right?

Comment (lrineau) [reply Number_types/include/CGAL/NT_converter.h:87]: Yes I think so. To be checked.

**Comment by sgiraudot** [comment]
Alright, so I removed the two apparently useless `NT_converter<NT,NT>` specializations (the existing `double/double` and the `float/float` I introduced with this PR). I run the testsuite of `Poisson` and `Number_types`, both succeed, so I think it's safe to say they were really useless. Let's see if it passes the whole testsuite.

**Comment by sloriot** [comment]
```
/home/runner/work/cgal/cgal/Cartesian_kernel/include/CGAL/Cartesian_converter.h:406:15: error: ambiguous template instantiation for ‘struct CGAL::NT_converter<double, double>’
     Converter c;
               ^
```

**Comment by sloriot** [comment]
@sgiraudot  this one is getting urgent as it is one of the last targetting 5.1.3

**Comment by sgiraudot** [comment]
Alright, I removed the commit that removed the overloads of `NT_converter<double,double>` and `NT_converter<float,float>`. It turns out these overloads are indeed useful because otherwise, compilers can't choose between:
- `NT_converter<NT, NT>`
- `NT_converter<NT, double>` (which also exists)

So these overloads _are_ indeed useful.

**Comment by sloriot** [comment]
`disable_if/enable_if` is also an option but not necessarily useful in this context

**Comment by lrineau** [comment]
> Alright, I removed the commit that removed the overloads of `NT_converter<double,double>` and `NT_converter<float,float>`. It turns out these overloads are indeed useful because otherwise, compilers can't choose between:
> 
>     * `NT_converter<NT, NT>`
> 
>     * `NT_converter<NT, double>` (which also exists)
> 
> 
> So these overloads _are_ indeed useful.

Then re-added them with a useful comment. :smile:

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.3-Ic-78.shtml

**Pull request merged**